test/workflow_test.go

```diff
@@ -3174,6 +3174,49 @@ func (w *Workflows) RunsLocalAndNonlocalActsWithRetries(ctx workflow.Context, nu
 	return nil
 }
 
+func (w *Workflows) SelectorBlockSignal(ctx workflow.Context, enableFlag bool) (string, error) {
+	ctx = workflow.WithActivityOptions(ctx, w.defaultActivityOptions())
+	var logger = workflow.GetLogger(ctx)
+	logger.Info("calling ExecuteActivity")
+	ch1 := workflow.NewChannel(ctx)
+	ch2 := workflow.NewChannel(ctx)
+
+	if enableFlag {
+		internal.SetUnblockSelectorSignal()
```

Comment:
I think we should do this as part of the test, not the workflow. More importantly though can you confirm setting this won't effect any test that runs after this test?

Reply:
> we should do this as part of the test, not the workflow

What is there difference between the two? or is this more of a style/preference?

> can you confirm setting this won't effect any test that runs after this test?

Looks like it does affect subsequent tests :( Is it enough to unset this value? Go tests run in parallel, so that seems insufficient?

Reply:
>What is there difference between the two? or is this more of a style/preference?

I think it is clearer when we are testing a workflow if all the setup is run before hand. 

> Go tests run in parallel, so that seems insufficient?

Only if [t.Parallel()](https://pkg.go.dev/testing#T.Parallel) is called